cmd/agent/subcommands/analyzelogs/command.go

```diff
@@ -63,22 +76,34 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			return fxutil.OneShot(runAnalyzeLogs,
 				core.Bundle(),
 				fx.Supply(cliParams),
-				fx.Supply(command.GetDefaultCoreBundleParams(cliParams.GlobalParams)),
+				fx.Supply(core.BundleParams{
+					ConfigParams: config.NewAgentParams(globalParams.ConfFilePath, config.WithFleetPoliciesDirPath(globalParams.FleetPoliciesDirPath)),
+					SecretParams: secrets.NewEnabledParams(),
+					LogParams:    log.ForOneShot("", "off", true)}),
+				dualTaggerfx.Module(common.DualTaggerParams()),
+				workloadmetafx.Module(defaults.DefaultParams()),
+				autodiscoveryimpl.Module(),
 			)
 		},
 	}
 
 	// Add flag for core config (optional)
 	cmd.Flags().StringVarP(&cliParams.CoreConfigPath, "core-config", "C", defaultCoreConfigPath, "Path to the core configuration file (optional)")
+	// Add flag for inactivity timeout (optional)
+	cmd.Flags().IntVarP(&cliParams.inactivityTimeout, "inactivity-timeout", "t", 1, "Time (seconds) that the program will wait for new logs before exiting (optional)")
 
 	return []*cobra.Command{cmd}
 }
 
 // runAnalyzeLogs initializes the launcher and sends the log config file path to the source provider.
-func runAnalyzeLogs(cliParams *CliParams, config config.Component) error {
-	outputChan, launchers, pipelineProvider := runAnalyzeLogsHelper(cliParams, config)
+func runAnalyzeLogs(cliParams *CliParams, config config.Component, ac autodiscovery.Component) error {
+	outputChan, launchers, pipelineProvider := runAnalyzeLogsHelper(cliParams, config, ac)
+	if outputChan == nil {
```

Comment:
Instead of checking if `outputChan` is `nil` and printing errors from inside `runAnalizeLogsHelper` it would be more go idiomatic to return an error as the last return value for the function and check `if err != nil` here and then print the contents of that error